ResearchKit/Common/ORKReviewStepViewController.m

```diff
@@ -0,0 +1,243 @@
+}
+
+- (void)setLearnMoreButtonItem:(UIBarButtonItem *)learnMoreButtonItem {
+    [super setLearnMoreButtonItem:learnMoreButtonItem];
+    _headerView.learnMoreButtonItem = self.learnMoreButtonItem;
+}
+
+- (void)setSkipButtonItem:(UIBarButtonItem *)skipButtonItem {
+    [super setSkipButtonItem:skipButtonItem];
+    _continueSkipView.skipButtonItem = self.skipButtonItem;
+}
+
+- (void)stepDidChange {
+    [super stepDidChange];
+    
+    [_tableContainer removeFromSuperview];
+    _tableContainer = nil;
+    
+    _tableView.delegate = nil;
+    _tableView.dataSource = nil;
+    _tableView = nil;
+    _headerView = nil;
+    _continueSkipView = nil;
+    
+    if ([self reviewStep]) {
+        _tableContainer = [[ORKTableContainerView alloc] initWithFrame:self.view.bounds];
+        _tableView = _tableContainer.tableView;
+        _tableView.delegate = self;
+        _tableView.dataSource = self;
+        _tableView.clipsToBounds = YES;
+
+        [self.view addSubview:_tableContainer];
+        _tableContainer.tapOffView = self.view;
+        
+        _headerView = _tableContainer.stepHeaderView;
+        _headerView.captionLabel.useSurveyMode = self.step.useSurveyMode;
+        _headerView.captionLabel.text = [self reviewStep].title;
+        _headerView.instructionLabel.text = [self reviewStep].text;
+        _headerView.learnMoreButtonItem = self.learnMoreButtonItem;
+        
+        _continueSkipView = _tableContainer.continueSkipContainerView;
+        _continueSkipView.skipButtonItem = self.skipButtonItem;
+        _continueSkipView.continueEnabled = YES;
+        _continueSkipView.continueButtonItem = self.continueButtonItem;
+        _continueSkipView.optional = self.step.optional;
+        [_tableContainer setNeedsLayout];
+    }
+}
+
+- (ORKReviewStep *)reviewStep {
+    return [self.step isKindOfClass:[ORKReviewStep class]] ? (ORKReviewStep *) self.step : nil;
+}
+
+//TODO: state restoration
+
+#pragma mark UITableViewDataSource
+
+- (NSInteger)numberOfSectionsInTableView:(nonnull UITableView *)tableView {
+    return _steps.count > 0 ? ORKReviewSectionCount : 0;
+}
+
+- (NSInteger)tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    if (section == ORKReviewSectionSpace1 || section == ORKReviewSectionSpace2) {
+        return 1;
+    }
+    NSMutableArray *textChoices = [[NSMutableArray alloc] init];
+    [_steps enumerateObjectsUsingBlock:^(id object, NSUInteger idx, BOOL *stop) {
+        ORKStep *step = (ORKStep*) object;
+        //TODO: process results
+        if (step.title) {
+            [textChoices addObject:@[step.title]];
+        }
+    }];
+    _answerFormat = [[ORKTextChoiceAnswerFormat alloc] initWithStyle:ORKChoiceAnswerStyleSingleChoice textChoices: textChoices];
+    _choiceCellGroup = [[ORKTextChoiceCellGroup alloc] initWithTextChoiceAnswerFormat:_answerFormat answer:nil beginningIndexPath:[NSIndexPath indexPathForRow:0 inSection:section] immediateNavigation:YES];
```

Comment:
`ORKTextChoiceCellGroup` really should not be used at here, an `ORKTextChoiceAnswerFormat` is not relevant too.
I suggest you create plain cells.